### PR TITLE
Fix Deprecation Warning for trim() Function in awkExistsAndRunsCorrectly()

### LINKED ISSUE #20959: [PHP 8.1 compatibility] Deprecated - trim(): Passing null to parameter #1 ($string) of type string
From: https://forum.matomo.org/t/probleme-de-recherche-de-mise-a-jour/51570/11
 - [ ] `/core/CliMulti/Process.php(259): Deprecated - trim(): Passing null to parameter #1 ($string) of type string is deprecated - Matomo 4.14.2 - Piwik\ErrorHandler::errorHandler(),#1/core/CliMulti/Process.php(259),#2/core/CliMulti/Process.php(244),#3/core/CliMulti/Process.php(200),#4/core/CliMulti.php(289),#5/core/CliMulti.php(85),#6/plugins/Diagnostics/Diagnostic/CronArchivingCheck.php(63),#7/plugins/Diagnostics/DiagnosticService.php(68),#8/plugins/Diagnostics/DiagnosticService.php(54),#9/plugins/Installation/Controller.php(525)`
 - [ ] `/core/CliMulti/Process.php(259): Deprecated - trim(): Passing null to parameter #1 ($string) of type string is deprecated - Matomo 4.14.2 - Piwik\ErrorHandler::errorHandler(),#1/core/CliMulti/Process.php(259),#2/core/CliMulti/Process.php(244),#3/plugins/Diagnostics/Diagnostic/CronArchivingCheck.php(68),#4/plugins/Diagnostics/DiagnosticService.php(68),#5/plugins/Diagnostics/DiagnosticService.php(54),#6/plugins/Installation/Controller.php(525),[internal function]: Piwik\Plugins\Installation\Controller->systemCheckPage(),#8/core/FrontController.php(637),#9/core/FrontController.php(169)`

I think this is not related to #20658

### PR DESCRIPTION
**Pull request by mindbreaker**
### Description:

This pull request addresses a deprecation warning that arises when a null value is passed to the trim() function in the awkExistsAndRunsCorrectly() method.

Changes:

-     Added a ternary check to ensure that the trim() function is only called if the output from the shell command is not null.
-     If the output is null, the function will now return false.

This change prevents the deprecation warning and ensures compatibility with the updated behavior of the trim() function in PHP.

fixes https://github.com/matomo-org/matomo/issues/20959

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: core/CliMulti/Process.php [hunk 1]
```diff
@@ -256,7 +256,7 @@ private static function psExistsAndRunsCorrectly()
     private static function awkExistsAndRunsCorrectly()
     {
         $testResult = @shell_exec('echo " 537 s000 Ss 0:00.05 login -pfl theuser /bin/bash -c exec -la bash /bin/bash" | ' . self::AWK_COMMAND . ' 2>/dev/null');
-        return trim($testResult) == '537';
+        return $testResult !== null ? trim($testResult) == '537' : false;
```
Comment (michalkleiner) [inline core/CliMulti/Process.php]:
```suggestion
        return trim($testResult ?? '') == '537';
```

**Comment by michalkleiner** [comment]
Thanks for the PR, @mindbreaker. The usual approach is to use the null-coalesce operator `??` that ensures a fallback value is provided when the tested value is null. Could you update the code as suggested? I'd be happy to merge it then.

**Review by michalkleiner — APPROVED** [review]

**Comment by sgiehl** [comment]
Thanks @mindbreaker for your contribution 🎉

**Comment by FabianSchmick** [comment]
Can this fix also please be merged into the current maintained Version 4.x ?

**Comment by sgiehl** [comment]
@FabianSchmick I've just cherry-picked that to 4.x, so it will be included in the next release there as well.

**Comment by FabianSchmick** [comment]
Great! Thank you

**Pull request merged**